### PR TITLE
AB#1255492 Support lists as inputs to ORTModule

### PR DESCRIPTION
**Pull request by baijumeswani**
**Description**: Add support for lists as inputs to ```ORTModule```.

### CODE REVIEW

File: orttraining/orttraining/python/training/ortmodule/_io.py [hunk 1]
```diff
@@ -129,8 +129,20 @@ def _combine_input_buffers_initializers(params, onnx_input_names, input_info, bu
         * Initializers: computed from original PyTorch model parameters
     '''
 
+    def _expand_inputs(current_input, non_none_inputs):
+        if isinstance(current_input, abc.Sequence):
```
Comment (mrry) [inline orttraining/orttraining/python/training/ortmodule/_io.py:133]: Do we need to exclude strings here, or is that covered by an earlier `isinstance`?

File: orttraining/orttraining/python/training/ortmodule/_io.py [hunk 2]
```diff
@@ -129,8 +129,20 @@ def _combine_input_buffers_initializers(params, onnx_input_names, input_info, bu
...
+                _expand_inputs(inp, non_none_inputs)
+        elif current_input is not None:
+            # else just collect all the non none inputs within non_none_inputs
+            non_none_inputs.append(current_input)
+
+
     # User inputs
-    non_none_inputs = [inp for inp in inputs if inp is not None]
+    non_none_inputs = []
+    _expand_inputs(inputs, non_none_inputs)
```
Comment (mrry) [inline orttraining/orttraining/python/training/ortmodule/_io.py:145]: What would it take to support list-valued arguments in `kwargs` as well?

Comment (baijumeswani) [reply orttraining/orttraining/python/training/ortmodule/_io.py:145]: Let me try to add support for ```kwargs``` argument being a list.

File: orttraining/orttraining/python/training/ortmodule/_io.py (hunk 1 above)
Comment (baijumeswani) [reply orttraining/orttraining/python/training/ortmodule/_io.py:133]: We don't need another ```isinstance``` for ```str``` checking here as it is covered earlier.

File: orttraining/orttraining/python/training/ortmodule/_io.py (hunk 2 above)
Comment (baijumeswani) [reply orttraining/orttraining/python/training/ortmodule/_io.py:145]: Support for ```kwargs``` might need more work than I had anticipated. I'll create another work item to address support for lists in ```kwargs```. For now, this PR unblocks some of the models that need lists as input to ```ORTModule```, so we could decouple the two.

**Comment by garymm** [comment]
Can you try adding `Fixes AB#1255492` to the pull request description or commit message? I want to see if it will work.

**Comment by baijumeswani** [comment]
> Can you try adding "Fixes [AB#1255492](https://msdata.visualstudio.com/3adb301f-9ede-41f2-933b-fcd1a486ff7f/_workitems/edit/1255492)" to the pull request description or commit message? I want to see if it will work.

Seems like it worked. 👍🏽

**Review by thiagocrepaldi — APPROVED** [review]

**Pull request merged**